ticket-management/ticket-management/Controllers/EventController.cs

```diff
@@ -1,99 +1,82 @@
-            {
-                EventDescription = e.EventDescription,
-                EventName = e.EventName,
-                EventType = e.EventType.EventTypeName,
-                Venue = e.Venue?.VenueLocation
-            });
-            */
-
-            var dtoEvents = _mapper.Map<List<EventDTO>>(events);
-
-            return Ok(dtoEvents);
+            return Ok(events);
         }
 
-
         [HttpGet]
-        public async Task<ActionResult<EventDTO>> GetById(long id)
+        public async Task<ActionResult<List<EventDTO>>> GetEvents([FromQuery] long? venueId, [FromQuery] string? eventTypeName)
         {
-            var @event = await _eventRepository.GetById(id);
-
-            if (@event == null)
+            if (venueId == null && eventTypeName == null)
             {
-                return NotFound();
+                var events = await _eventService.GetAll();
+                return Ok(events);
             }
-
-            /*
-            var dtoEvent = new EventDTO()
+            else if (venueId != null && eventTypeName == null)
             {
-                EventDescription = @event.EventDescription,
-                EventName = @event.EventName,
-                EventType = @event.EventType.EventTypeName,
-                Venue = @event.Venue?.VenueLocation
-            };
-            */
-
-            var dtoEvent = _mapper.Map<EventDTO>(@event);
-
-            return Ok(dtoEvent);
+                var events = await _eventService.GetAllByVenue(venueId.Value);
+                return Ok(events);
+            }
+            else if (venueId == null && eventTypeName != null)
+            {
+                var events = await _eventService.GetAllByType(eventTypeName);
+                return Ok(events);
+            }
+            else if (venueId != null && eventTypeName != null)
+            {
+                var events = await _eventService.GetAllByVenueAndType(venueId.Value, eventTypeName);
+                return Ok(events);
+            }
+            else
+            {
+                var events = new List<EventDTO>();
+                return Ok(events);
+            }
         }
 
+
         [HttpPatch]
         public async Task<ActionResult<EventPatchDTO>> Patch(EventPatchDTO eventPatch)
         {
-            var eventEntity = await _eventRepository.GetById(eventPatch.EventId);
+            var result = await _eventService.Update(eventPatch);
 
-            if (eventEntity == null)
+            if (!result)
             {
                 return NotFound();
             }
 
-            if (!string.IsNullOrEmpty(eventPatch.EventName)) eventEntity.EventName = eventPatch.EventName;
-            if (!string.IsNullOrEmpty(eventPatch.EventDescription)) eventEntity.EventDescription = eventPatch.EventDescription;
-
-            _eventRepository.Update(eventEntity);
-
             return NoContent();
         }
 
         [HttpDelete]
         public async Task<ActionResult> Delete(long id)
         {
```

Comment:
as face putin diferit delete-ul in general . in metoda din service, sa ai direct delete (daca ajungi in service.delete inseamna ca acel item chiar exista). iar aici in controller apelezi mai intai get (id) din service, daca e null, dai not found, otherwise apelezi delete. operatiunile din service sa fie cat mai basic (sa nu faci si get si delete in aceeasi metoda), si astea sa le faci doar in controller (ca se presupune ca un controller e mai smart, nu e asa basic :D )